parastell/utils.py

```diff
@@ -311,3 +312,178 @@ def combine_dagmc_models(models_to_merge):
+    counter-clockwise direction"""
+    angle = np.deg2rad(angle)
+    rotation_mat = np.array(
+        [
+            [np.cos(angle), -np.sin(angle), 0],
+            [np.sin(angle), np.cos(angle), 0],
+            [0, 0, 1],
+        ]
+    )
+    ribs = np.dot(ribs, rotation_mat.T)
+    return ribs
+
+
+def ribs_from_kisslinger_format(filename, start_line=2, scale=m2cm):
+    """Reads a Kisslinger format file and returns a list of toroidal angles,
+    along with a list of lists of the rib loci (in x,y,z) at each toroidal
+    angle. It is expected that toroidal angles are provided in degrees.
+
+    Arguments:
+        filename (str): Path to the file to be read.
+        start_line (int): Line at which the data should start being read,
+            should skip any comments. Defaults to 2.
+        scale (float): Amount to scale the r-z coordinates by. Defaults to 100.
+    Returns:
+        toroidal_angles (list of float): List of all the toroidal angles in the
+            angles in the input file in degrees.
+        profiles (numpy array): 3 dimensional numpy array where the first
+            dimension corresponds to individual ribs, the second to the
+            position on a rib, and the third to the actual x,y,z coordinate
+            of the locus.
+    """
+
+    with open(file=filename) as file:
+        data = file.readlines()
+
+    profiles = []
+    profile = []
+    toroidal_angles = []
+
+    for line in data[start_line - 1 :]:
+        if line.count("\t") == 2:
+            num_toroidal_angles, num_poloidal_angles, periods = (
+                line.rstrip().split("\t")
+            )
+        elif "\t" not in line and "#" not in line:
+            toroidal_angle = float(line.rstrip())
+            toroidal_angles.append(toroidal_angle)
+            if len(profile) != 0:
+                profiles.append(profile)
+                profile = []
+        else:
+            line = line.rstrip()
+            r_z_coords = [float(coord) * scale for coord in line.split("\t")]
+            x_coord = r_z_coords[0] * np.cos(np.deg2rad(toroidal_angle))
+            y_coord = r_z_coords[0] * np.sin(np.deg2rad(toroidal_angle))
+            z_coord = r_z_coords[1]
+            profile.append([x_coord, y_coord, z_coord])
+    profiles.append(profile)
+
+    return (
+        np.array(toroidal_angles),
+        num_toroidal_angles,
+        num_poloidal_angles,
+        periods,
+        np.array(profiles),
+    )
+
+
+def extract_rib_data(
+    ribs, toroidal_angles, poloidal_angles, x_data, y_data, z_data, grid_points
+):
+    for phi, rib in zip(toroidal_angles, ribs):
+        for theta, rib_locus in zip(poloidal_angles, rib):
+            x_data.append(rib_locus[0])
+            y_data.append(rib_locus[1])
+            z_data.append(rib_locus[2])
+            grid_points.append([phi, theta])
+    return x_data, y_data, z_data, grid_points
+
+
+class KisslingerSurface(object):
```

Comment:
Should this be in IVB somewhere as a variation on the current surface definition to allow us to use them more interchangeably?  I haven't really thought about the details, but assumed that's how this would come together, similar to what you've done with magnet definitions